File: src/main/java/io/dustin/finstage/content/dto/ContentListRequest.java
package io.dustin.finstage.content.dto;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class ContentListRequest {

    private String symbol;

    @Min(1)
    private int page = 1;

    @Min(1)
    @Max(100)
    private int size = 20;

    public static ContentListRequest of(String symbol, int page, int size) {
        ContentListRequest request = new ContentListRequest();
        request.symbol = symbol;
        request.page = page;
        request.size = size;
        return request;
    }
}

File: src/main/kotlin/io/dustin/finstage/content/dto/ContentListRequest.kt
package io.dustin.finstage.content.dto

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

class ContentListRequest private constructor(
        var symbol: String?,

        @field:Min(1)
        var page: Int = 1,

        @field:Min(1)
        @field:Max(100)
        var size: Int = 20
) {
    companion object {
        fun of(symbol: String?, page: Int, size: Int): ContentListRequest {
            return ContentListRequest(symbol, page, size)
        }
    }
}
